src/utilities/construct_randoms_from_GEsingles.cxx

```diff
@@ -87,6 +87,11 @@ int main(int argc, char **argv)
   GE::RDF_HDF5::SinglesRatesFromGEHDF5  singles(input_filename);
   const float coincidence_time_window = input_file.get_coincidence_time_window();
 
-  randoms_from_singles(proj_data, singles, coincidence_time_window);
+  // If half-life exists in proj_data, use that value, otherwise get it from the input_file
+  const float isotope_halflife = proj_data.get_exam_info().get_radionuclide().get_half_life(false) != -1
+                                 ? proj_data.get_exam_info().get_radionuclide().get_half_life()
+                                 : input_file.get_halflife();
```

Comment:
Are we therefore dependant on a fix of #962?

Reply:
> I think this illistrates an issue with current code because of how projdata is constructed when a template is provided.
> ...
>  I also realise that this is becoming a construct_randoms_from_singles redesign

yes sure. Let's create an issue for this but not fix that here

> Is there any reason to keep the template sinogram option for the construct_randoms_from_GEsingles utility,

yes, to be able to specify different time frames and/or span/mashing.

> Are we therefore dependant on a fix of #962?

I wouldn't fix that here either...

My suggestion is just to optionally pass half-life, and if it isn't passed, get it from the `projdata`. But in the utility, we always pass it.